src/DiffViewerProvider.ts

```diff
@@ -37,24 +29,38 @@ export class DiffViewerProvider implements vscode.CustomReadonlyEditorProvider<D
       enableScripts: true,
     };
 
-    const config = this.extractConfig();
-    const diffFiles = parse(diffDocument.content, config);
+    const updateWebview = () => {
```

Comment:
IMO we can move this entire function outside `resolveCustomTextEditor` as a private method.
WDYT?

Reply:
Yeah, pulling the code out of `resolveCustomTextEditor` would be nice. 

Please correct me if I'm wrong, as I'm new to VSCode extensions, but it seems to me that the private method could even be static. If that's true, I'll want to check your opinion on a bigger question: private static methods vs module functions.

I feel that module functions are better than private static methods because a) there's less nesting, and b) also less typing. How would you feel about a top-level function like this?

```js
function updateWebview(diffDocument: vscode.TextDocument; webviewPanel: vscode.WebviewPanel): void
```

I realize I've removed your `args`. Was that intended so that inside `resolveCustomTextEditor` we could just store `diffDocument` and `webviewPanel` in a single object that we would pass to `updateWebview()` in both places where we call it? If so, I feel that the extra parameter in each invocation is OK, compared to creating and remembering another object. 

As it happens, I currently have time for discussions like this. If you don't, please just say so and I'll try to do what you say.

Reply:
Yes, it is perfectly fine to have private static methods. 
However, static methods belong to the class not the instance.
In our case, `updateWebview` seems to belong to instances of `DiffViewerProvider`.

I personally tend to organize the code following OOP.
In JS/TS code, this avoids having loose functions spread over the files.
(saying this because my background includes many years of java development)

Although there is no example in this repository, nowadays I also tend to use an `args` object in functions that have more than one argument as this favor auto-completion and avoids issues, especially if the function arguments are the same type. 

I really enjoy discussions like this but can't promise quick replies. :)

Reply:
Understood, I've committed a new version, is that what you had in mind?

Though I disagree on static private methods vs module functions, it's your project so I'll try to match your style. I too have done years of Java, but I don't miss it. Still, it's not a huge difference anyway. 8-)